tests/func/test_get.py

```diff
@@ -252,3 +236,50 @@ def test_get_pipeline_tracked_outs(
     with git_dir.chdir():
         Repo.get("file:///{}".format(os.fspath(tmp_dir)), "bar", out="baz")
         assert (git_dir / "baz").read_text() == "foo"
+
+
+def make_subrepo(dir_, scm, config):
+    dir_.mkdir(parents=True)
+    with dir_.chdir():
+        dir_.scm = scm
+        dir_.init(dvc=True, subdir=True)
+        dir_.add_remote(config=config)
+
+
+@pytest.mark.parametrize(
+    "output",
+    [
+        "foo",
+        {"foo": "foo", "bar": "bar"},
+        {"subdir": {"foo": "foo", "bar": "bar"}},
+    ],
+    ids=["file", "dir", "nested_dir"],
+)
+@pytest.mark.parametrize(
+    "erepo", [pytest.lazy_fixture("erepo_dir"), pytest.lazy_fixture("git_dir")]
+)
+@pytest.mark.parametrize(
+    "subrepo_paths",
+    [
+        (os.path.join("sub", "subdir1"), os.path.join("sub", "subdir2")),
+        (os.path.join("sub"), os.path.join("sub", "subdir1")),
+    ],
+    ids=["isolated", "nested"],
+)
+def test_subrepo_multiple(
+    tmp_dir, scm, output, subrepo_paths, erepo, local_cloud
+):
+    sub_repos = [erepo / path for path in subrepo_paths]
+    filename = "output"
+    for repo in sub_repos:
+        make_subrepo(repo, erepo.scm, local_cloud.config)
+        repo.dvc_gen({filename: output}, commit="add subrepo")
+        repo.dvc.push()
+
+    for i, repo in enumerate(sub_repos):
+        Repo.get(
+            f"file:///{erepo}",
```

Comment:
Should we provide local URL with `file:///` prefix? It seems to me that it is passed straight to `external_repo` and we never actually call `_fix_upstream`, because `os.path.isdir` returns False.

Reply:
I think the intention was to avoid git's optimization, see `--local` option on [`git clone`](https://git-scm.com/docs/git-clone#_options). And, `local_cloud.config` is a temp dir, meaning it's an abspath, so it's fine.

But, by doing this, we might miss local git repos, so, I'll rollback (or, try using mix of both in tests).